src/hera/workflows/script.py

```diff
@@ -434,25 +439,19 @@ def _get_outputs_from_parameter_annotations(
     artifacts: List[Artifact] = []
 
     for name, p in inspect.signature(source).parameters.items():
-        annotation = get_workflow_annotation(p.annotation)
-        if not annotation or not annotation.output:
+        annotation = construct_io_from_annotation(name, p.annotation)
```

Comment:
seems better to have tests for the changes in `src/hera/workflows/script.py` (`_get_outputs_from_parameter_annotations` and `_extract_all_output_annotations`)
even if someone updates `construct_io_from_annotation`, we can make sure this is working as expected.